syntax = "proto3";

package model.inference;

service InferenceService {
  rpc Infer(InferRequest) returns (InferResponse);
}

message InferRequest {
  bytes image_bytes = 1;
  float confidence_threshold = 2;
  string bbox_format = 3;
  bool include_segmentation = 4;
}

message RGB {
  uint32 r = 1;
  uint32 g = 2;
  uint32 b = 3;
}

message DefectInfo {
  string name = 1;
  int32 class_id = 2;
  RGB ui_color = 3;
  RGB mask_color = 4;
}

message BBox {
  float x = 1;
  float y = 2;
  float w = 3;
  float h = 4;

  string label = 5;
  int32 class_id = 6;
  float confidence = 7;
}

message InferResponse {
  string model_name = 1;
  repeated BBox list_bbox = 2;
  optional bytes img_segmentation = 3; // ausente em detecção
  repeated DefectInfo defect_list = 4;
  string error = 5;
}
